sfepy/terms/terms_diffusion.py

```diff
@@ -479,3 +479,59 @@ class AdvectDivFreeTerm(ScalarDotMGradScalarTerm):
     arg_shapes = {'material' : 'D, 1', 'virtual' : ('1', 'state'),
                   'state' : '1'}
     mode = 'grad_state'
+
+
+class NonlinearDiffusionTerm(Term):
+    """
+     Product of gradient of virtual, gradient of state and function of state.
```

Comment:
This might be better?

The diffusion term with a scalar coefficient given by a user supplied function dependent on the state variable.

Has it any sense to generalize also for f(p) returning a dim x dim matrix?